composer.json

```diff
@@ -7,10 +7,13 @@
     },
     "require": {
         "php": "^5.6 || ^7.0 || ^8.0",
-        "guzzlehttp/guzzle": "^6.0 || ^7.0",
+        "psr/http-client-implementation": "^1.0",
+        "psr/http-factory-implementation": "^1.0",
         "paragonie/random_compat": "^1 || ^2 || ^9.99"
     },
     "require-dev": {
+        "guzzlehttp/guzzle": "^7.3",
+        "guzzlehttp/psr7": "^2.0@RC",
```

Comment:
For tests, uses RC because this package doesn't declare providing the PSR17 factories until 2.x.